### PR TITLE
Changes required for Spring Insight on PCF 1.5.x

### PR DESCRIPTION
**Pull request by dtillman**
This commit makes a number of changes which are required for
compatibility with Spring Insight on PCF 1.5.x.

### CODE REVIEW

**Comment by cgfrost** [comment]
Hi Dave, this looks good to me but could you update the pull request to remove the commented out code. If it's not needed I'd rather not have it there at all.

Thanks, Chris.

**Comment by dtillman** [comment]
Hi Chris,
I've removed the commented out old code!
Dave

**Comment by cgfrost** [comment]
Looks good, I think @nebhale want to have a chat with you before I merge it in though. I'm sure he will get in touch soon.

**Comment by cgfrost** [comment]
Hi Dave, I'm waiting to get the OK to merge this in but I'd like to keep it as ready as possible in the mean time. We're about to do a release and I want to avoid hold ups when it comes time to do it. So, with that in mind could you re-base on top of master and squash the pull request in to a single commit. While you are at it could you check the indentation hasn't changed. Rubocop will pull you up on it if it has during a `bundle exec rake` run. I've commented the code where it looks like it might of changed.

File: spec/java_buildpack/framework/spring_insight_spec.rb [hunk 1]
```diff
@@ -31,51 +31,51 @@
...
+      expect(sandbox + 'weaver/insight-weaver-2.0.0-CI-SNAPSHOT.jar').to exist
       expect(sandbox + 'insight/conf/insight.properties').to exist
+      expect(sandbox + 'insight/agent-plugins/insight-agent-rabbitmq-core-2.0.0-CI-SNAPSHOT.jar').to exist
     end
 
-    it 'guarantees that internet access is available when downloading' do
+    it 'does guarantee that internet access is available when downloading' do
       expect_any_instance_of(JavaBuildpack::Util::Cache::InternetAvailability)
-        .to receive(:available).with(true, 'The Spring Insight download location is always accessible')
+      .to receive(:available).with(true, 'The Spring Insight download location is always accessible')
```
Comment (cgfrost) [inline spec/java_buildpack/framework/spring_insight_spec.rb]: Indentation.

File: lib/java_buildpack/framework/spring_insight.rb [hunk 2]
```diff
@@ -45,28 +45,23 @@ def detect
...
-          .add_system_property('org.aspectj.tracing.factory', 'default')
-          .add_system_property('insight.transport.type', 'HTTP')
-
-        add_agent_configuration
+        .add_javaagent(weaver_jar)
+        .add_system_property('insight.base', insight_directory)
+        .add_system_property('insight.logs', logs_directory)
+        .add_system_property('aspectj.overweaving', true)
+        .add_system_property('org.aspectj.tracing.factory', 'default')
       end
```
Comment (cgfrost) [inline lib/java_buildpack/framework/spring_insight.rb]: Indentation.

**Comment by cgfrost** [comment]
Hi Dave, I'm OK now to merge this in and get it released once you have done the changes. Thanks, Chris.

**Pull request closed without merging**